src/qibo/config.py

```diff
@@ -88,6 +89,20 @@ def set_batch_size(batch_size):
     SHOT_BATCH_SIZE = batch_size
 
 
+def get_metropolis_threshold():
+    """Returns threshold for using Metropolis algorithm for sampling measurement shots."""
+    return SHOT_METROPOLIS_THRESHOLD
+
+def set_metropolis_threshold(threshold):
+    """Sets threshold for using Metropolis algorithm for sampling measurement shots."""
+    if not isinstance(threshold, int):
+        raise_error(TypeError, "Shot threshold must be integer.")
+    elif threshold < 1:
+        raise_error(ValueError, "Shot threshold be a positive integer.")
+    global SHOT_METROPOLIS_THRESHOLD
+    SHOT_METROPOLIS_THRESHOLD = threshold
```

Comment:
If I understand correctly, these last 2 functions are required in order to update the `SHOT_METROPOLIS_THRESHOLD` dynamically during the code execution, correct? So far, however the code doesn't use this feature anywhere, so this is a feature for the user, right?

Reply:
No, this is used in the `sample_frequencies` function of the qibotf backend [here](https://github.com/qiboteam/qibo/blob/48928bf1873cd7a067dcc2fc7b8ff2a7eb4ec599/src/qibo/backends/tensorflow.py#L227). If the number of shots requested by the user is smaller than this number then sampling falls back to tf primitives (`tf.random.categorical`) while if it is larger the Metropolis custom operator is used.

Reply:
Sorry, I was not clear, I mean set/get_metropolis_threshold() functions. These functions are not used anywhere in the code, I believe.

Reply:
Oh yes, that's correct, it is just for the user to use if they want to disable the Metropolis (by setting very high threshold) for some reason (for example if it does not produce good results for their distribution). I think this is the case with most set/get functions we already have, like `set_precision`, `set_batch_size`, etc., they are not used anywhere in the code.

Reply:
Ok, thanks for confirmations. We might include some instructions about that in the "Advanced examples" section.